File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/04.For-Loop/03-additional/01.backToThePast.js
function backToThePast(input) {

    let age = 18;
    const initialYear = 1800;
    const costPerYear = 12000;

    const inheriredMoney = Number(input[0]);
    const endYear = Number(input[1]);

    let moneyNeeded = 0;

    for (let i = initialYear; i <= endYear; i++) {
        if (i % 2 === 0) {
            moneyNeeded += costPerYear;
        } else {
            moneyNeeded += costPerYear + 50 * age;
        }

        age += 1;
    }

    const difference = Math.abs(inheriredMoney - moneyNeeded).toFixed(2);
    if (moneyNeeded <= inheriredMoney) {
        console.log(`Yes! He will live a carefree life and will have ${difference} dollars left.`);
    } else {
        console.log(`He will need ${difference} dollars to survive.`);
    }
}
backToThePast(["100000.15", "1808"]);

File: Python_Basics/Homeworks-And-Labs/04.for_loop/03.additional/01.back_to_the_past.py
age = 18
initial_year = 1800
cost_per_year = 12000

inherited_money = float(input())
end_year = int(input())

money_needed = 0

for year in range(initial_year, end_year + 1):
    if year % 2 == 0:
        money_needed += cost_per_year
    else:
        money_needed += cost_per_year + 50 * age
    age += 1

difference = abs(inherited_money - money_needed)
if money_needed <= inherited_money:
    print(f'Yes! He will live a carefree life and will have {difference:.2f} dollars left.')
else:
    print(f'He will need {difference:.2f} dollars to survive.')
